src/js/redux/reducers.js

```diff
@@ -54,7 +54,7 @@ export function appNavClick(state, { payload }) {
         globalNav: payload.custom ? state.globalNav && state.globalNav.map(item => ({
             ...item,
             active: payload && (item.id === payload.id || item.title === payload.id)
-                || (item.subItems && item.subItems.some(({ id }) => id === payload.id))
+                || (item.subItems && item.subItems.some(({ id }) => id === payload.id) && item.id === state.appId)
```

Comment:
Can you add payload parent here as well?

```suggestion
                || (item.subItems && item.subItems.some(({ id }) => id === payload.id) && (item.id === state.appId || item.id === payload.parentId))
```